tests/DCPS/LivelinessTest/LivelinessTest.cpp

```diff
@@ -0,0 +1,564 @@
+#include "common.h"
+#include "Writer.h"
+#include "DataReaderListener.h"
+
+#include <tests/DCPS/common/TestException.h>
+#include <tests/DCPS/FooType4/FooDefTypeSupportImpl.h>
+
+#include <dds/DCPS/Service_Participant.h>
+#include <dds/DCPS/Marked_Default_Qos.h>
+#include <dds/DCPS/Qos_Helper.h>
+#include <dds/DCPS/TopicDescriptionImpl.h>
+#include <dds/DCPS/PublisherImpl.h>
+#include <dds/DCPS/SubscriberImpl.h>
+#include <dds/DCPS/transport/framework/TransportRegistry.h>
+#include <dds/DCPS/StaticIncludes.h>
+#include <dds/DCPS/WaitSet.h>
+#if defined ACE_AS_STATIC_LIBS && !defined OPENDDS_SAFETY_PROFILE
+#include <dds/DCPS/transport/udp/Udp.h>
+#endif
+
+#include <dds/DdsDcpsSubscriptionC.h>
+
+#include <ace/Arg_Shifter.h>
+#include <ace/Reactor.h>
+#include <ace/OS_NS_unistd.h>
+
+class ReactorCtrl : public ACE_Event_Handler
+{
+public:
+  ReactorCtrl() : cond_(lock_) {}
+
+  int handle_timeout (const ACE_Time_Value &tv,
+                      const void *arg)
+  {
+    ACE_UNUSED_ARG(tv);
+    ACE_UNUSED_ARG(arg);
+
+    // it appears that you must have the lock before waiting or signaling on Win32
+    ACE_GUARD_RETURN (ACE_Recursive_Thread_Mutex, guard, this->lock_, -1);
+
+    return cond_.wait();
+  }
+
+private:
+  ACE_Recursive_Thread_Mutex lock_;
+  ACE_Condition<ACE_Recursive_Thread_Mutex> cond_;
+} ;
```

Comment:
```suggestion
#include <ace/OS_NS_unistd.h>
```

This seems to have been used in `publisher.cpp`, but looks like it's not used anymore.